table/tables/cache.go

```diff
@@ -42,8 +44,8 @@ type cachedTable struct {
 	cacheData atomic.Value
 	handle    StateRemote
 	totalSize int64
-
-	renewReadLease tokenLimit
+	// StateRemote is not thread-safe, this tokenLimit is used to keep only one visitor.
+	tokenLimit
```

Comment:
Just a suggestion... Maybe we can force users to use tokenLimit before using `StateRemote` like:

```
type tokenLimit = chan StateRemote
```

and provide methods:

```
func TakeStateRemoteHandle() StateRemote {
  handle := <- tokenLimit
  return handle
}

func TakeStateRemoteHandleNoWait() StateRemote {
    select {
    case handle := <-tokenLimit:
       return handle
    default:
      return nil
    }
}

func PutRemotehandle(handle StateRemote) {
  tokenLimit <- handle
}
```

It can achieve the thread safe goal and easier to use

Reply:
This code looks smart, I'll take a try

Reply:
Done